src/features/weak-references.md

```diff
@@ -177,7 +177,7 @@ Eventually, after a garbage collection occurs, the `MovingAvg` instance and the
 But there’s still a problem here: we’ve added a level of indirection to `listener` by wrapping it a `WeakRef`, but the wrapper in `addWeakListener` is still leaking for the same reason that `listener` was leaking originally. Granted, this is a smaller leak since only the wrapper is leaking instead of the whole `MovingAvg` instance, but it is still a leak. The solution to this is the companion feature to `WeakRef`, `FinalizationRegistry`. With the new `FinalizationRegistry` API, we can register a callback to run when the garbage collector zaps a register object. Such callbacks are known as _finalizers_.
 
 :::note
-**Note:** The finalization callback does not run immediately after garbage-collecting the event listener. It either runs at some point in the future, or not at all — the spec doesn’t guarantee that it runs! Keep this in mind when writing code.
+**Note:** The finalization callback does not run immediately after garbage-collecting the event listener, so don't use it for important logic or metrics. The timing of garbage collection and finalization callbacks is unspecifed. In fact, an engine that never garbage collected would be fully compliant. However, it's safe to assume that engines will garbage collect, and finalization callbacks will be called at _some point_. Keep this uncertainty in mind when writing code.
```

Comment:
"and finalization callbacks will be called at _some point_" is a dangerous suggestion to make, since they might not run if there's no need to worry about memory, e.g. the tab is closed.

Reply:
Looks like this is resolved, marking as such.